pkg/cmd/roachtest/tests/restore.go

```diff
@@ -414,23 +414,97 @@ func (tpccIncData) runRestoreDetached(
+// until after tpce init ended.
+//
+// Fixture recreation steps:
+//
+// 1) Create a roachprod cluster with the same topology as
+// this test.
+//
+// 2) Setup the cluster to run tpc-e init with 500k customers, using
+// the repro steps in the appendix of Nathan's v22.2
+// Scalability & Efficiency Evaluation using TPC-E doc
+// https://docs.google.com/document/d/1wzkBXaA3Ap_daMV1oY1AhQqlnAjO3pIVLZTXY53m0Xk/edit
+//
+// 3) Set the gc ttl for the tpce database to 25 hrs
+//
+// 4) Create a backup schedule with revision history and hourly incremental backups.
+//
+// 6) Run the tpce workload for 24 hours with 100k active customers.
+type tpce10TB struct{}
+
+func (tpce10TB) name() string {
+	return "TPCC10TB"
+}
+
+func (tpce10TB) runRestore(ctx context.Context, c cluster.Cluster) {
+	// Restore from the first full backup AOST in the last incremental backup in the chain.
+	c.Run(ctx, c.Node(1), `./cockroach sql --insecure -e "
+				RESTORE FROM '/2022/11/06-124208.52' IN
+				'gs://cockroach-fixtures/backups/tpc-e/rev-history=true,inc-count=23,cluster/customers=500k/22.1.8?AUTH=implicit'
+				AS OF SYSTEM TIME '2022-11-06 23:40:22'"`)
+}
+
+func (tpce10TB) runRestoreDetached(
+	ctx context.Context, t test.Test, c cluster.Cluster,
+) (jobspb.JobID, error) {
+	// Restore from the first full backup AOST in the last incremental backup in the chain.
+	c.Run(ctx, c.Node(1), `./cockroach sql --insecure -e "
+				RESTORE FROM '/2022/11/06-124208.52' IN
+				'gs://cockroach-fixtures/backups/tpc-e/rev-history=true,inc-count=23,cluster/customers=500k/22.1.8?AUTH=implicit'
+				AS OF SYSTEM TIME '2022-11-06 23:40:22' WITH detached"`)
+	db, err := c.ConnE(ctx, t.L(), c.Node(1)[0])
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to connect to node 1; running restore detached")
+	}
+
+	var jobID jobspb.JobID
+	if err := db.QueryRow(`SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'RESTORE'`).Scan(&jobID); err != nil {
+		return 0, err
+	}
+
+	return jobID, nil
+}
+
 var _ testDataSet = tpccIncData{}
 
 func registerRestore(r registry.Registry) {
-	largeVolumeSize := 2500 // the size in GB of disks in large volume configs
 
 	for _, item := range []struct {
-		nodes        int
-		cpus         int
-		largeVolumes bool
+		nodes int
+		cpus  int
+
+		// pdVolumeSize specifies, in GB, the pd-ssd GB per node. If not specified,
+		// each node will use a local ssd, which on GCP, has 375 GB capacity.
+		pdVolumeSize int
 		dataSet      testDataSet
 
 		timeout time.Duration
 	}{
 		{dataSet: dataBank2TB{}, nodes: 10, timeout: 6 * time.Hour},
 		{dataSet: dataBank2TB{}, nodes: 32, timeout: 3 * time.Hour},
-		{dataSet: dataBank2TB{}, nodes: 6, timeout: 4 * time.Hour, cpus: 8, largeVolumes: true},
+		{dataSet: dataBank2TB{}, nodes: 6, timeout: 4 * time.Hour, cpus: 8, pdVolumeSize: 2500},
 		{dataSet: tpccIncData{}, nodes: 10, timeout: 6 * time.Hour},
+		{dataSet: tpce10TB{}, nodes: 10, timeout: 10 * time.Hour, cpus: 16, pdVolumeSize: 1500},
```

Comment:
Is the CPUs=16 intentional? The reason we upped the CPUs on the n6, pd2500 configuration is that we use that as a signal to up the restore and addsstable concurreny in the roachtest below.

Reply:
ah, i bumped cpus to run the test on bigger machines, so the restore would run more quickly.  I did not realize it would affect these cluster settings. I could try running this test with the default cpu size (4) or remove that logic from the loop by adding a pre test hook field to the struct that contains all the tests.

Reply:
using 4vcpu’s, this [restore](http://35.231.108.106:26258/#/overview/list) , is really really slow,
- Hardware metrics indicate we’re cpu saturated
- Disk Usage/ Total ranges indicates there’s 12 MB per range
- Memory profiles on nodes look similar to the original restore split [issue](https://github.com/cockroachdb/cockroach/issues/86470)
- hardware metrics indicate we’re writing about 20 Mib/node/s. roachperf indicates we usually have 6x throughput
- Also the merge queue has been capped at 100k for several hours.

Reply:
i wonder how this restore would fare with the target size set to 384... Hopefully the 10x improvement[ david saw](https://cockroachlabs.slack.com/archives/C03JCUUSCD6/p1666708399408599) a few weeks ago on :D

Reply:
runs fine on 8vcpus in 4 or so hours!